interp/src/stk_env.rs

```diff
@@ -810,16 +821,24 @@ impl<K: Eq + std::hash::Hash, V: Eq> Smoosher<K, V> {
             }
         }
 
-        a = a.smoosh(dp_first - 1);
+        a = a.smoosh(dp_first.unwrap() - 1);
 
         let mut a_head = a.head;
 
         //iterate over every smooshed smoosher and put all of their values in
         //the head of the first smoosher.
         for sm in smooshed {
             for (k, v) in sm.head {
-                if a_head.insert(k, v).is_some() {
-                    panic!("arguments of merge are not disjoint");
+                if let Some(prev) = a_head.get(&k) {
+                    // overlap accepable for defined keys as long as they agree
+                    if overlap_keys.contains(&k) && prev == &v {
```

Comment:
Does this check if conflicting assignments to the same port have the same source port and allows that?

Reply:
It's not that smart, it's just checking that they have the same value. The source isn't stored w/ assignments and wouldn't be known at this layer since it's generic over keys and values. This is perhaps an argument to de-generic the stack environment.